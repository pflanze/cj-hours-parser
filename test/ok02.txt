# this is a comment<!-- with an internal comment-->
[
this is a multiline
  com<!--with more internal commenting-->ment
]
-12:44:21 test  some time
-17:16:31UTC time for some stuff
-17:16UTC time for some stuff

[[
some more commenting
]] 
#[and yet more]
[
blabla
]
TZ=Zurich
03:27-6:44:01 test-again
 TZ           = london


  set 	 hello = 123

  set 	 hello2 123too

Mon_31_Oct_200350_CET_2022-{Mon 31 Oct 21:18:04 GMT 2022}infrastructure
-23:01 project-foo
03:27-10:44:10GMT<!--foo--> test-again
-{Wed 14 Dec 23:42:51 CET 2022}something
-{Sun Oct 30 15:43:04 CET 2022}aufstellen
-16:39
03:27-10:44:10GMT<foo> test-again
