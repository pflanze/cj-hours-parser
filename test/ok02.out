item=SingleLineComment(Owned(" this is a comment"))
    =SingleLineComment(Owned(" this is a comment"))
item=MultiLineComment([Owned("this is a multiline"), Inline("  comment")])
    =MultiLineComment([Owned("this is a multiline"), Inline("  comment")])
item=Other([Minus, Time(NT(12:44:21)), Rest(Inline("test  some time"))])
    =Other([Subitem(L(0, 6, 0), Minus), Subitem(L(0, 6, 1), Time(NT(12:44:21))), Subitem(L(0, 6, 10), Rest(Inline("test  some time")))])
item=Other([Minus, Time(NTZ(17:16:31, Zone(Inline("UTC")))), Rest(Owned("time for some stuff"))])
    =Other([Subitem(L(0, 7, 0), Minus), Subitem(L(0, 7, 1), Time(NTZ(17:16:31, Zone(Inline("UTC"))))), Subitem(L(0, 7, 13), Rest(Owned("time for some stuff")))])
item=Other([Minus, Time(NTZ(17:16:00, Zone(Inline("UTC")))), Rest(Owned("time for some stuff"))])
    =Other([Subitem(L(0, 8, 0), Minus), Subitem(L(0, 8, 1), Time(NTZ(17:16:00, Zone(Inline("UTC"))))), Subitem(L(0, 8, 10), Rest(Owned("time for some stuff")))])
item=Empty(1)
    =Empty(1)
item=MultiLineComment([Owned("some more commenting")])
    =MultiLineComment([Owned("some more commenting")])
item=SingleLineComment(Inline("[and yet more]"))
    =SingleLineComment(Inline("[and yet more]"))
item=MultiLineComment([Inline("blabla")])
    =MultiLineComment([Inline("blabla")])
item=ProcessingInstruction { key: Inline("TZ"), value: Inline("Zurich") }
    =ProcessingInstruction { key: Inline("TZ"), value: Inline("Zurich") }
item=Other([Time(NT(03:27:00)), Minus, Time(NT(06:44:01)), Rest(Inline("test-again"))])
    =Other([Subitem(L(0, 18, 0), Time(NT(03:27:00))), Subitem(L(0, 18, 5), Minus), Subitem(L(0, 18, 6), Time(NT(06:44:01))), Subitem(L(0, 18, 14), Rest(Inline("test-again")))])
item=ProcessingInstruction { key: Inline("TZ"), value: Inline("london") }
    =ProcessingInstruction { key: Inline("TZ"), value: Inline("london") }
item=Empty(2)
    =Empty(2)
item=ProcessingInstruction { key: Inline("hello"), value: Inline("123") }
    =ProcessingInstruction { key: Inline("hello"), value: Inline("123") }
item=Empty(1)
    =Empty(1)
item=ProcessingInstruction { key: Inline("hello2"), value: Inline("123too") }
    =ProcessingInstruction { key: Inline("hello2"), value: Inline("123too") }
item=Empty(1)
    =Empty(1)
item=Other([Rest(Owned("Mon_31_Oct_200350_CET_2022")), Minus, Time(NT(21:18:04)), Rest(Inline("infrastructure"))])
    =Other([Subitem(L(0, 26, 0), Rest(Owned("Mon_31_Oct_200350_CET_2022"))), Subitem(L(0, 26, 26), Minus), Subitem(L(0, 26, 28), Time(NT(21:18:04))), Subitem(L(0, 26, 57), Rest(Inline("infrastructure")))])
item=Other([Minus, Time(NT(23:01:00)), Rest(Inline("project-foo"))])
    =Other([Subitem(L(0, 27, 0), Minus), Subitem(L(0, 27, 1), Time(NT(23:01:00))), Subitem(L(0, 27, 7), Rest(Inline("project-foo")))])
item=Other([Time(NT(03:27:00)), Minus, Time(NTZ(10:44:10, Zone(Inline("GMT")))), Rest(Inline("test-again"))])
    =Other([Subitem(L(0, 28, 0), Time(NT(03:27:00))), Subitem(L(0, 28, 5), Minus), Subitem(L(0, 28, 6), Time(NTZ(10:44:10, Zone(Inline("GMT"))))), Subitem(L(0, 28, 28), Rest(Inline("test-again")))])
item=Other([Minus, Time(NT(23:42:51)), Rest(Inline("something"))])
    =Other([Subitem(L(0, 29, 0), Minus), Subitem(L(0, 29, 2), Time(NT(23:42:51))), Subitem(L(0, 29, 31), Rest(Inline("something")))])
item=Other([Minus, Time(NT(15:43:04)), Rest(Inline("aufstellen"))])
    =Other([Subitem(L(0, 30, 0), Minus), Subitem(L(0, 30, 2), Time(NT(15:43:04))), Subitem(L(0, 30, 31), Rest(Inline("aufstellen")))])
item=Other([Minus, Time(NT(16:39:00))])
    =Other([Subitem(L(0, 31, 0), Minus), Subitem(L(0, 31, 1), Time(NT(16:39:00)))])
item=Other([Time(NT(03:27:00)), Minus, Time(NTZ(10:44:10, Zone(Inline("GMT")))), Rest(Owned("<foo> test-again"))])
    =Other([Subitem(L(0, 32, 0), Time(NT(03:27:00))), Subitem(L(0, 32, 5), Minus), Subitem(L(0, 32, 6), Time(NTZ(10:44:10, Zone(Inline("GMT"))))), Subitem(L(0, 32, 17), Rest(Owned("<foo> test-again")))])
Hello, world!
